portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,32 +14,74 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import com.google.gson.Gson;
-import java.util.List;
-import java.util.Arrays;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-
 public class DataServlet extends HttpServlet {
-  
-  private List<String> groceryList;
-
-  public DataServlet(){  
-    groceryList = Arrays.asList("Fruit", "Soap", "Cereal");
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(groceryList);
-    response.setContentType("application/json;");
+    final String commentTable = "Comments";
+    final String commentCol = "comments";
+    final String timestampCol = "timestamp";
+
+    final Query commentQuery = new Query(commentTable).addSort(timestampCol, SortDirection.DESCENDING);
```

Comment:
This is not a correct way of using "final".

It's used to prevent a method be overridden or a class to be subclassed.